packages/platforms/browser/lib/auto-instrumentation/resource-load-plugin.ts

```diff
@@ -54,9 +54,22 @@ export class ResourceLoadPlugin implements Plugin<BrowserConfiguration> {
         const parentContext = this.spanContextStorage.first
 
         if (parentContext) {
-          const url = new URL(entry.name)
-          url.search = ''
-          const name = url.href
+          const networkRequestInfo = configuration.networkRequestCallback({ url: entry.name, type: entry.initiatorType })
+
+          if (!networkRequestInfo) return
+
+          if (typeof networkRequestInfo.url !== 'string') {
+            configuration.logger.warn(`expected url to be a string following network request callback, got ${typeof networkRequestInfo.url}`); return
+          }
+
+          let name = ''
+          try {
+            const url = new URL(networkRequestInfo.url)
+            url.search = ''
+            name = url.href
+          } catch (err) {
+            configuration.logger.warn('unable to parse url'); return
```

Comment:
@gareththackeray would you know what type of wording to use in this scenario? where a user has returned a string from the `networkRequestCallback` but we are unable to construct a URL with it.

Reply:
How about:
> Unable to parse URL returned from networkRequestCallback.

And include the URL in the message?